www/pfSense-pkg-vHosts/files/usr/local/pkg/vhosts.inc

```diff
@@ -0,0 +1,367 @@
+    {
+    worker_connections  1024;
+    }
+
+http
+    {
+    include             /usr/local/etc/nginx/mime.types;
+    default_type        application/octet-stream;
+    add_header          X-Frame-Options SAMEORIGIN;
+    server_tokens       off;
+    sendfile            on;
+
+    access_log          syslog:server=unix:/var/run/log,facility=local5 combined;
+    keepalive_timeout   75;
+
+    @servers
+    }
+CONF;
+
+/*--------------------------------------------*/
+/* nginx server config block template.        */
+/*                                            */
+/* @listen_conf = 'listen' directive(s).      */
+/* @hostnames   = Host names.                 */
+/* @ssl_conf    = Secure configuration block. */
+/* @root_folder = Document root folder.       */
+/*--------------------------------------------*/
+$vhosts_g['conf_server'] = <<<CONF_SERVER
+
+    server
+        {
+        @listen_conf
+        server_name @hostnames;
+        @ssl_conf
+
+        client_max_body_size 200m;
+
+        gzip on;
+        gzip_types text/plain text/css text/javascript application/x-javascript text/xml application/xml application/xml+rss application/json;
+
+        root '@root_folder';
+
+        @custom_config
+
+        location / { index  index.html index.htm index.php; }
+
+        location ~ \.php$
+            {
+            try_files            \$uri =404; # This line closes a potential security hole
+                                             # ensuring users can't execute uploaded files
+                                             # see: http://forum.nginx.org/read.php?2,88845,page=3
+            fastcgi_pass         unix:/var/run/php-fpm.socket;
+            fastcgi_index        index.php;
+            fastcgi_param        SCRIPT_FILENAME  \$document_root\$fastcgi_script_name;
+            fastcgi_param        HTTP_PROXY  ""; # Fix httpoxy - https://httpoxy.org/#fix-now
+            fastcgi_read_timeout 180;
+            include              /usr/local/etc/nginx/fastcgi_params;
+            }
+
+        location = /favicon.ico  # Do not log requests for favicon.ico.
+            {
+            return               204;
+            access_log           off;
+            log_not_found        off;
+            }
+        }
+CONF_SERVER;
+
+/*-----------------------------------------*/
+/* nginx server SSL config block template. */
+/*                                         */
+/* @crt_file_path = Certificate file path. */
+/* @key_file_path = Private key file path. */
+/*-----------------------------------------*/
+$vhosts_g['conf_server_ssl'] = <<<CONF_SERVER_SSL
+
+        ssl_certificate                         '@crt_file_path';
+        ssl_certificate_key                     '@key_file_path';
+        ssl_session_timeout                     10m;
+        keepalive_timeout                       70;
+        ssl_session_cache                       shared:SSL:10m;
+        ssl_protocols                           TLSv1 TLSv1.1 TLSv1.2;
```

Comment:
I disagree. The primary purpose of this tool was to provide a way to put up a server down page for a server behind the router undergoing maintenance. By turning off TLSv1, I would be restricting that server to TLSv1.1 or higher. That is not my decision to make. But to disable it in the future, with an option to enable, is certainly reasonable.